pkg/agent/config/config.go

```diff
@@ -348,3 +350,14 @@ func getConfig(info *clientaccess.Info) (*config.Control, error) {
 	controlControl := &config.Control{}
 	return controlControl, json.Unmarshal(data, controlControl)
 }
+
+func hostnameCheck(hostname string) {
+	for {
+		_, err := sysnet.LookupHost(hostname)
+		if err == nil {
+			break
+		}
+		logrus.Infof("Waiting for hostname %s to be resolvable: %v", hostname, err)
+		time.Sleep(2 * time.Second)
+	}
+}
```

Comment:
This seems to loop forever. Would it be better to fail after 30 tries (~ 1 minute)?